### ISSUE TITLE
[UX] Rename "taxonomy" to "Categorization" in the user interface

### ISSUE THREAD

**Issue by jenlampton** [backdrop/backdrop-issues#203]
When people want to categorize their content, they don't think to themselves, "I want to add a system of taxonomy to my website", they think "how do I add categories?".  We need to make our user-interfaces match the mental model of these users.

I propose that we change the word "Taxonomy" in the user interface to "Categories" everywhere it appears.  I believe this was what it said back in Drupal 5 (or was it 4.7) and people had a much easier time with it back then.

Also up for debate: Is it worth renaming "Vocabularies" and "Terms" or would that do more harm than good?

Brainstorming of Issues
1. Taxonomy
A. English speakers do not regularly use or understand the word "taxonomy"; it should be avoided wherever possible in Drupal.
B. "Taxonomy" is largely equivalent to: classification, ontology [yes, a word even worse than "taxonomy"], categorization, organization, etc.
2. Vocabulary
A. English speakers do not regularly use or understand the word "vocabulary" in any sense other than one's use of words in a given language; it should be avoided wherever possible in Drupal.
B. "Vocabulary" is largely equivalent to: category, order, group, set, division, class, structure, cluster, hierarchy, tree, graph, network, etc.
3. Term
A. "Term" isn't terrible but is also used rarely in English; when it is used, "term" frequently signifies time rather than an element.
B. "Term" is largely equivalent to: tag, label, node [this goes to show how undistinctive, i.e. useless, "node" is], mnemonic, key, keyword, element, item, designation, etc.

Principles of Assessment
1. What is most intuitive to the beginning user?
2. Intermediate and expert users will adapt to any terminology that labels a set of features and is sufficiently described or discoverable, but for beginners the terminology is very important to the learning curve.
3. Don't worry about the exact definitions for any particular word: worry about perception, first impressions, and standard practice.

Let's Look to History (aka, there is nothing new under the sun)
- [Initial creation of taxonomy.module](http://cvs.drupal.org/viewvc.py/drupal/drupal/modules/taxonomy.module?revision=1.1&view=markup) (Apr. 2002)
- [Should we rename "taxnomy" to "classification"?](http://lists.drupal.org/archives/drupal-user/2003-06/msg00061.html) (Jun. 2003)
- [Renamed 'taxonomy' menu to 'categories' for sake of usability](http://cvs.drupal.org/viewvc.py/drupal/drupal/modules/taxonomy/taxonomy.module?revision=1.110&view=markup). (Apr. 2004)
- [Drupal terminology suggestions](http://drupal.org/node/48576) (Feb. 2006)
- [Taxonomy module terminology](https://drupal.org/node/192209) (Feb. 2006)

**Comment by docwilmot** [comment]
We should do this.

**Comment by klonos** [comment]
When I see any of the two words, I can't help my Greek nature kicking in. They both make sense to me as they would to any Greek person. I guess what I'm trying to say is that I don't feel strong about any one word in particular.

**Comment by sutibun** [comment]
I remember when I first got into Drupal, I didn't understand taxonomy and I like to think I am fairly technical. **Only because I once participated in a biology-related internship years before that I kind of got an inkling.** I had to create a few vocabularies, itself another weird name, and play around with a few dummy examples to see if I understood correctly.

Fast forward to a few months ago while building a Drupal site for someone else (someone I don't consider a dumb average user)... and I didn't know how to explain taxonomy and vocabulary. They understood what a category and tag was, just not the rest.

Suggestions
----
Anyway, I don't claim to have the answer but I thought about the way Drupal/Backdrop goes about it

- Taxonomy > Vocabulary > Tags
- Taxonomy > Vocabulary > Category

and some alternate words came up such as Organize, Catalog, Label, Sort, File, Index, Classify, Catalog, Peg, Designate, Assign.

And that got me thinking.

So, here are some ideas I am throwing out there. They sound less high falutin and more down to earth.

Sort of.

I just don't know if they make sense as I think it does.

### Idea 1

- Label > Catalog type > Tag
- Label > Catalog type > Category

### Idea 2

- Organize content > Label type > Category
- Organize content > Label type > Tags

### Thinking behind this

I like the word "Label".

When I think of label, I think of labeling boxes. We can even use visuals to help give context (a box label icon or maybe even a pricing gun) 

- In the *physical* world, you need to "have a Labeling system in place to organize merchandise."
- In the *online* world, you need to "Label your content with tags and categories."

That's my line of thinking.

If I had to choose, I like "Organize content > Label type > Tags" best.

What do you guys think?

**Comment by klonos** [comment]
> I like the word "Label".

The only thing with "label" is that we already use it in other, *important* places. Predominantly for field labels. That would complicate things instead of simplifying them I'm afraid.

**Comment by klonos** [comment]
...one thing I've noticed is that we use both "tags" and "terms" in the UI/menu:

![backdrop-tags_vs_terms](https://cloud.githubusercontent.com/assets/2423362/9702468/f9bbbf26-54a1-11e5-94a3-f6b96d5c80cd.png)

We should stick to one. If we keep "vocabularies", then I believe "terms" is more suitable than tags. If we decide to change to something like "categories", then I think "tags" fits better.

What do you think about this:

Structure -> Tag content -> Tag groups -> Tags

**Comment by klonos** [comment]
...or:

Structure -> Content tags -> Tag groups -> Tags

in place of:

Structure -> Taxonomy -> Vocabularies -> Terms

I think that it'd be easier to grasp and remember that:

- there are "tags"
- you can use them to *tag* content (assign tags to content)
- you can *group* them in "tag groups"

So "tags", "tag" (the verb) and "tag group"! It feels as if the three/four terms ("taxonomy", "vocabularies" and "tags"/"terms") are reduced to only one. People only have to remember the word "tag". :wink:

**Comment by Graham-72** [comment]
:+1: 
So much easier to say too.

**Comment by quicksketch** [comment]
> ...one thing I've noticed is that we use both "tags" and "terms" in the UI/menu:

It's not that we use both, but that "Tags" is the default vocabulary that comes with the standard.profile. You can delete the Tags vocabulary if you so choose.

I do like that this would simplify down the terminology a bit.

- Term => Tag
- Vocabulary => Tag group

Though "Tag group" seems a little bit of a strange name, I think users would be able to grasp it more quickly than a "Vocabulary" or "Taxonomy".

**Comment by sutibun** [comment]
Replacing Taxonomy
------

**I actually like "Tag content".**  (the verb)

If you're looking to add tags to your content, the fact "tag" is in "Tag content" makes it obvious where to go as opposed to "Taxonomy", which is like another language to the majority of users.

Though I will add, **at first sight I wasn't sure what "Tag content" meant as a whole**. I had to think about it a little. I initially took the first word as a noun. Maybe "Tag your content" so it's clearer we're talking about the verb?

Replacing Vocabulary
----

I don't know about "tag groups" though. Still can't wrap my head around that. How about "Tag type"?

I like creating sentences using these very words to see if we can make sense of them.

> In Backdrop, we can **create content** like an **Article or a Page** BUT, at the same time, we can create our own **type of content** (content types) if we choose to.

Maybe we can have a similar line of thinking here.

> In Backdrop, we can **tag content** with a **Category or a Tag** BUT, at the same time, we can create our own **type of tags** (tag type), our custom tagging system,  if we choose to.

How does "tag type" sound?

### Other alternatives for vocabulary

These also came up. Dumping them here as it might help spark an idea

- Tag content > Tagging system
- Tag content > Labeling system
- Tag content > Classification system 
- Tag content > categorization type (not sure I like this one)
- Label your content > Label type > Marker (or Ticket?)

They're not great but more indicative than "Vocabulary". 

Replacing Term
------

To replace Term, maybe "Tag element" or "Tag marker"? I'm not sure here.

We can't use Term here since that only makes "sense" when talking about Vocabulary"

Summary
-----

In other words, instead of 

- "Taxonomy > Vocabulary > Term"

we'd have:

- "Tag content > Tag type > Tag element" OR
- "Tag your content > Tag type > Tag marker"

Of course, if we can have that, I suppose we can also consider:

- "Categorize content > Category type > Category element

On a sidenote, whatever we choose, how about adding descriptive text on the Taxonomy page saying something along the lines of 

> "Just like you apply labels on boxes to identify, sort, & organize them, so too can you label your content with tags and categories to better group and organize them."

**Comment by BWPanda** [comment]
Is this discussion still relevant for 1.x? If so, I like 'tags' and 'tag groups' (or 'tag types').

I'm also wondering if we can introduce the new terminology in 1.x as changes just to the human labels/menu items (while keeping the existing terminology in the machine names/URLs), and then change everything to the new terminology in 2.x...?

**Comment by jenlampton** [comment]
Changing strings breaks translations. This means D7 translations and Backdrop 1.x translations will need to be updated to work on newer versions.

Changing URLs is now safe in 1.x, thanks to `backdrop_goto_deprecated()`.

If I were looking for a way to categorize my content, I wouldn't look for something called `Taxonomy` and I'm not sire I'd look for something called `Tags` either. 

I'd prefer
Taxonomy => Categorization
Vocabulary => Category Type / Group / Set / Collection
Term => Category

Another thought: We are already using `Tag` as a field label so that might get confusing if we also had tags as a structural item.

**Comment by klonos** [comment]
🤔

**Comment by jenlampton** [comment]
This old PR had way to many merge conflicts...  removing label. 

Naming things is hard, let's [push the renaming of vocabularies and terms](https://github.com/backdrop/backdrop-issues/issues/4222) to 2.x, but change Taxonomy now. Maybe it will be easier to get a consensus?
